lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -3500,6 +3507,10 @@ getNetworkInformation st nl = liftIO $ do
         , Api.nodeTip = apiNodeTip
         , Api.networkTip = fst <$> nowInfo
         , Api.nodeEra = toApiEra nodeEra
+        , Api.networkInfo =
+            Api.ApiNetworkInfo
+                (if nid == Cardano.Mainnet then "mainnet" else "testnet")
```

Comment:
Hi @paolino 

If anyone ever adds a new constructor for `ProtocolMagic` (impossible to predict), then wouldn't this conversion break?

Perhaps a more robust option would be to define a function that takes a `ProtocolMagic` and returns the string you want, where that function pattern matches on all of the constructors explicitly.

See https://github.com/input-output-hk/adrestia/blob/master/docs/code/Coding-Standards.md#prefer-pattern-matching-to-equality-testing-on-sum-types

Reply:
I changed it to use the constructors from Cardano.Api, would this cover your point?